folium/map.py

```diff
@@ -21,7 +21,26 @@
 )
 
 
-class Layer(MacroElement):
+class Evented(MacroElement):
+    """The base class for Layer and Map
+
+    Adds the `on` method for event handling capabilities.
+
+    See https://leafletjs.com/reference.html#evented for
+    more in depth documentation. Please note that we have
+    only added the `on(<Object> eventMap)` variant of this
+    method using python keyword arguments.
+    """
+
+    def __init__(self):
+        super().__init__()
+
```

Comment:
this init is not necessary I think, without it it will just go directly to the init of the parent class

```suggestion
```